### PR TITLE
Bug fixes for Pool Resize

### PR DESCRIPTION
**Pull request by anshuman-goel**
## Summary of the Pull Request
Mainly 2 bug fixes:
- No need to check for task state as we are already fetching tasks from available states.
- Not to delete all the free nodes. Instead delete only how much is determined before.

_What is this about?_
Other fixes:
- Sometimes nodes go into halt and delete themselves w/o reducing size of scaleset, so making sure scaleset maintains the size as required.
- Take into the account for the nodes which are marked as delete_requested.

## PR Checklist
* [x] Applies to work item: #124 
* [x] CLA signed. If not, go over [here](https://cla.opensource.microsoft.com/microsoft/onefuzz) and sign the CLI.
* [ ] Tests added/passed
* [ ] Requires documentation to be updated
* [ ] I've discussed this with core contributors already. If not checked, I'm ready to accept this work might be rejected in favor of a different grand plan. Issue number where discussion took place: #xxx

## Info on Pull Request
There is one more weird state that I face which I look at later on is, node is in init state but also requested for delete.

_What does this include?_

## Validation Steps Performed

_How does someone test & validate?_
Modify api.py to enable autoscaling. Then create and pool and jobs and then delete them.

### CODE REVIEW

File: src/api-service/__app__/onefuzzlib/autoscale.py [hunk 1]
```diff
@@ -1,8 +1,3 @@
-#!/usr/bin/env python
```
Comment (bmc-msft) [inline src/api-service/__app__/onefuzzlib/autoscale.py:1]: please re-add the copyright lines

**Review by bmc-msft — CHANGES_REQUESTED** [review]

**Comment by bmc-msft** [comment]
Testing in-progress

**Comment by bmc-msft** [comment]
Existing tests pass.  Once the copyright lines are added, I will merge.

File: src/api-service/__app__/onefuzzlib/autoscale.py (hunk 1 above)
Comment (anshuman-goel) [reply src/api-service/__app__/onefuzzlib/autoscale.py]: added

**Review by bmc-msft — APPROVED** [review]

**Pull request merged**